### ISSUE TITLE
Does not work with magento commerce 2.4

### ISSUE THREAD

**Issue by samc2810** [Smile-SA/magento2-module-elasticsuite-quick-order#3]
Environment
- PHP7.4
- ES 7.6.1
- Magento Commerce Cloud 2.4
- Elasticsuite latest

Hi guys
I just installed the elasticsuite quick order on my M2.4 project. Everything else is working as expected but the quickorder suggest
Are there any plans to upgade this module to be compatible with the latest Magento version? We just need to make a decision to use Elasticsuite or not for our current project and it would depend on the fact if you would support the Magento2.4 in the near future

Thank you

![image](https://user-images.githubusercontent.com/1092318/95839542-6bf32980-0d43-11eb-8e09-3b92b21ae00b.png)

Error Log

> 
> 1 exception(s):
> Exception #0 (Magento\Framework\Exception\RuntimeException): Type Error occurred when creating object: Smile\ElasticsuiteQuickOrder\Model\ResourceModel\Product\Suggest
> 
> Exception #0 (Magento\Framework\Exception\RuntimeException): Type Error occurred when creating object: Smile\ElasticsuiteQuickOrder\Model\ResourceModel\Product\Suggest
> <pre>#1 Magento\Framework\ObjectManager\Factory\Dynamic\Developer->create() called at [vendor/magento/framework/ObjectManager/ObjectManager.php:70]
> #2 Magento\Framework\ObjectManager\ObjectManager->get() called at [vendor/magento/framework/ObjectManager/Factory/AbstractFactory.php:167]
> #3 Magento\Framework\ObjectManager\Factory\AbstractFactory->resolveArgument() called at [vendor/magento/framework/ObjectManager/Factory/AbstractFactory.php:273]
> #4 Magento\Framework\ObjectManager\Factory\AbstractFactory->getResolvedArgument() called at [vendor/magento/framework/ObjectManager/Factory/AbstractFactory.php:236]
> #5 Magento\Framework\ObjectManager\Factory\AbstractFactory->resolveArgumentsInRuntime() called at [vendor/magento/framework/ObjectManager/Factory/Dynamic/Developer.php:34]
> #6 Magento\Framework\ObjectManager\Factory\Dynamic\Developer->_resolveArguments() called at [vendor/magento/framework/ObjectManager/Factory/Dynamic/Developer.php:59]
> #7 Magento\Framework\ObjectManager\Factory\Dynamic\Developer->create() called at [vendor/magento/framework/ObjectManager/ObjectManager.php:70]
> #8 Magento\Framework\ObjectManager\ObjectManager->get() called at [vendor/magento/framework/ObjectManager/Factory/AbstractFactory.php:167]
> #9 Magento\Framework\ObjectManager\Factory\AbstractFactory->resolveArgument() called at [vendor/magento/framework/ObjectManager/Factory/AbstractFactory.php:273]
> #10 Magento\Framework\ObjectManager\Factory\AbstractFactory->getResolvedArgument() called at [vendor/magento/framework/ObjectManager/Factory/AbstractFactory.php:236]
> #11 Magento\Framework\ObjectManager\Factory\AbstractFactory->resolveArgumentsInRuntime() called at [vendor/magento/framework/ObjectManager/Factory/Dynamic/Developer.php:34]
> #12 Magento\Framework\ObjectManager\Factory\Dynamic\Developer->_resolveArguments() called at [vendor/magento/framework/ObjectManager/Factory/Dynamic/Developer.php:59]
> #13 Magento\Framework\ObjectManager\Factory\Dynamic\Developer->create() called at [vendor/magento/framework/ObjectManager/ObjectManager.php:56]
> #14 Magento\Framework\ObjectManager\ObjectManager->create() called at [vendor/magento/framework/App/ActionFactory.php:44]
> #15 Magento\Framework\App\ActionFactory->create() called at [vendor/magento/framework/App/Router/Base.php:306]
> #16 Magento\Framework\App\Router\Base->matchAction() called at [vendor/magento/framework/App/Router/Base.php:167]
> #17 Magento\Framework\App\Router\Base->match() called at [vendor/magento/framework/Interception/Interceptor.php:58]
> #18 Magento\Framework\App\Router\Base\Interceptor->___callParent() called at [vendor/magento/framework/Interception/Interceptor.php:138]
> #19 Magento\Framework\App\Router\Base\Interceptor->Magento\Framework\Interception\{closure}() called at [vendor/magento/framework/Interception/Interceptor.php:153]
> #20 Magento\Framework\App\Router\Base\Interceptor->___callPlugins() called at [generated/code/Magento/Framework/App/Router/Base/Interceptor.php:26]
> #21 Magento\Framework\App\Router\Base\Interceptor->match() called at [vendor/magento/framework/App/FrontController.php:115]
> #22 Magento\Framework\App\FrontController->dispatch() called at [vendor/magento/framework/Interception/Interceptor.php:58]
> #23 Magento\Framework\App\FrontController\Interceptor->___callParent() called at [vendor/magento/framework/Interception/Interceptor.php:138]
> #24 Magento\Framework\App\FrontController\Interceptor->Magento\Framework\Interception\{closure}() called at [vendor/magento/module-store/App/FrontController/Plugin/RequestPreprocessor.php:99]
> #25 Magento\Store\App\FrontController\Plugin\RequestPreprocessor->aroundDispatch() called at [vendor/magento/framework/Interception/Interceptor.php:135]
> #26 Magento\Framework\App\FrontController\Interceptor->Magento\Framework\Interception\{closure}() called at [vendor/fastly/magento2/Model/FrontControllerPlugin.php:131]
> #27 Fastly\Cdn\Model\FrontControllerPlugin->aroundDispatch() called at [vendor/magento/framework/Interception/Interceptor.php:135]
> #28 Magento\Framework\App\FrontController\Interceptor->Magento\Framework\Interception\{closure}() called at [vendor/magento/module-page-cache/Model/App/FrontController/BuiltinPlugin.php:71]
> #29 Magento\PageCache\Model\App\FrontController\BuiltinPlugin->aroundDispatch() called at [vendor/magento/framework/Interception/Interceptor.php:135]
> #30 Magento\Framework\App\FrontController\Interceptor->Magento\Framework\Interception\{closure}() called at [vendor/magento/framework/Interception/Interceptor.php:153]
> #31 Magento\Framework\App\FrontController\Interceptor->___callPlugins() called at [generated/code/Magento/Framework/App/FrontController/Interceptor.php:26]
> #32 Magento\Framework\App\FrontController\Interceptor->dispatch() called at [vendor/magento/framework/App/Http.php:116]
> #33 Magento\Framework\App\Http->launch() called at [generated/code/Magento/Framework/App/Http/Interceptor.php:24]
> #34 Magento\Framework\App\Http\Interceptor->launch() called at [vendor/magento/framework/App/Bootstrap.php:263]
> #35 Magento\Framework\App\Bootstrap->run() called at [pub/index.php:40]
> </pre>

**Comment by romainruaud** [comment]
Yes we plan to support it, of course.

But I cannot give you a timeline for this. Maybe you can propose a Pull Request fixing this issue ?

**Comment by davidtisn** [comment]
Is it possible to get a patch for this? Right now we have issue compiling for 2.4:

```
public function __construct(
        \Magento\QuickOrder\Model\CatalogPermissions\Permissions $permissions,
        \Magento\Framework\DB\Helper $dbHelper,
        Visibility $catalogProductVisibility = null,
        SearchResultApplierInterfaceFactory $searchResultApplierInterfaceFactory = null
    ) {
        $this->permissions = $permissions;
        $this->dbHelper    = $dbHelper;

        parent::__construct($permissions, $dbHelper, $catalogProductVisibility, $searchResultApplierInterfaceFactory);
    }
```

**Comment by rbayet** [comment]
Hello @davidtisn,

The issue has been fixed in release 2.10.**1** (we now have a branch/release strategy mirroring that of Elasticsuite.)

Please update your version of the package or change your composer requirements with `composer require smile/module-elasticsuite-quick-order ~2.10.0`

Regards,

**Comment by davidtisn** [comment]
thanks.
